File: go-backend/blockchain/blockchain.go
package blockchain

import "time"

var Blockchain []Block

func InitGenesisBlock() Block {
	genesis := Block{
		Index:        0,
		Timestamp:    time.Now(),
		Transactions: nil,
		PrevHash:     "0",
		Hash:         "genesis_hash",
		Proof:        "GENESIS",
	}
	Blockchain = append(Blockchain, genesis)
	return genesis
}

func AddBlock(newBlock Block) {
	Blockchain = append(Blockchain, newBlock)
}

func GenerateBlock(prevBlock Block, transactions []Transaction, proof string) Block {
	newBlock := Block{
		Index:        prevBlock.Index + 1,
		Timestamp:    time.Now(),
		Transactions: transactions,
		PrevHash:     prevBlock.Hash,
		Proof:        proof,
	}

	newBlock.Hash = CalculateHash(newBlock)
	return newBlock
}

File: next-frontend/src/types/blockchain.ts
// filepath: c:\ProjectCPE\next-frontend\src\types\blockchain.ts
export interface Transaction {
    sender: string;
    receiver: string;
    amount: number;
    image_hash: string;
  }
  
  export interface Block {
    index: number;
    timestamp: string;
    transactions: Transaction[] | null;
    prev_hash: string;
    hash: string;
    proof: string;
  }